site/src/pages/release-notes/1.29.0.mdx

```diff
@@ -0,0 +1,248 @@
+---
+date: 2024-06-05
+---
+
+## 🌟 New features
+
+- You can now use an <type://AsyncLoader> to load a value atomically and asynchronously. #5506 #5590
+  ```java
+  Function<String, CompletableFuture<String>> loader = cache -> {
+    ResponseEntity<String> response =
+      client.prepare().get("/api/v1/items").asString().execute();
+      return response.thenApply(res -> res.content());
+  };
+
+  AsyncLoader<String> asyncLoader =
+    AsyncLoader
+      .builder(loader)
+      .expireAfterLoad(Duration.ofSeconds(60))
+      .build();
+  asyncLoader.get().join();
+  ```
+- Armeria [HTTP/JSON to GRPC transcoding](https://cloud.google.com/endpoints/docs/grpc/transcoding) service now
+  supports [Custom methods](https://cloud.google.com/apis/design/custom_methods). #5613
+  ```protobuf
+  rpc Watch(WatchRequest) returns (WatchResponse) {
+    option (google.api.http) = {
+      post: "/v1:watch" // 👈👈👈
+      body: "*"
+    };
+  }
+  ```
+- You can now specify when to remove multipart temporary files using <type://MultipartRemovalStrategy>. #5652 #5653
+  ```java
+  Server
+    .builder()
+    .multipartRemovalStrategy(MultipartRemovalStrategy.ON_RESPONSE_COMPLETION)
+    ...
+  ```
+- A <type://ClientConnectionTimings> now includes the time spent on the TLS handshake. #3647 #5647
+  ```java
+  ClientConnectionTimings timings = ...
+  timings.tlsHandshakeDurationNanos();
+  ```
+- You can now configure <type://TlsEngineType> using <type://ClientFactoryBuilder> or <type://ServerBuilder>. #4962
+  ```java
+  ClientFactory
+    .builder()
+    .tlsEngineType(TlsEngineType.OPENSSL)
+    .build();
+  ```
+- You can now easily find both dynamic and static decorators that handle a request
+  via <type://ServiceRequestContext.findService(Class)>. #5670
+  ```java
+  ServerBuilder sb = ...
+  sb.service("/foo", fooService);
+  ...
+  ServiceRequestContext ctx = ...
+  assert ctx.findService(FooService.class) == fooService;
+  ```
+- You can now use the marshaller specified by gRPC `MethodDescriptor`. #5103 #5630
+  ```java
+  GrpcClientBuilder builder = ...
+  builder.useMethodMarshaller(true);
+  ```
+- You can now programmatically retrieve the server side metrics via <type://ServerMetrics>. #4992 #5627
+  ```java
+  ServerMetrics metrics = serverConfig.serverMetrics();
+  metrics.activeConnections();
+  metrics.pendingRequests();
+  metrics.activeRequests();
+  ```
+- You can now use a <type://CoroutineHttpService> that runs your service in a coroutine scope. #5442 #5603
+  ```kotlin
+  ServerBuilder sb = ...
+  sb.service(
+    "/hello",
+    CoroutineHttpService { ctx, req ->
+      HttpResponse.of("hello world")
+    })
+  ```
+- You can now specify options for an <type://HttpService> via <type://ServiceOptions>. #5071 #5574
+  ```java
+  HttpService httpService = new HttpService() {
+    ...
+    @Override
+    public ServiceOptions options() {
+      return customizedOptions;
```

Comment:
Please make this example more useful. How about defining a static final field of `ServiceOptions`?